src/world/Explosion.php

```diff
@@ -214,11 +251,25 @@ public function explodeB() : bool{
 				}
 				$this->world->setBlockAt($pos->x, $pos->y, $pos->z, $airBlock);
 			}
+
+			if(isset($this->fireIgnitions[$hash]) &&
+				$block->getSide(Facing::DOWN)->getSupportType(Facing::UP) === SupportType::FULL
+			){
+				$this->world->setBlockAt($pos->x, $pos->y, $pos->z, $fireBlock);
```

Comment:
You are still calling setBlockAt() twice. Do this check in the place where air block is being set (on line 252) and set fire instead if the check is true.